lib/index.d.ts

```diff
@@ -56,18 +56,22 @@ export interface ServiceRequest {
     };
 }
 
-export interface ServiceClientResponse extends ResponseObject {
-    readonly payload:
-        | stream.Readable
+type ServiceClientResponsePayload = stream.Readable
         | Buffer
         | string
         | {[key: string]: string | string[]};
+
+export interface ServiceClientResponse extends ResponseObject {
+    readonly payload: ServiceClientResponsePayload;
 }
 
 export type ClientInstance = {
-    request: (
+    request: <T = ServiceClientResponsePayload>(
         serviceClientOptions: ServiceClientOptions
-    ) => ServiceClientResponse;
+    ) => T extends ServiceClientResponsePayload ? Promise<ServiceClientResponse> : Http.IncomingMessage & {
```

Comment:
This is still breaking, even if it fits the larger types scheme you're trying to inject; you will need to remove the conditional type inference.

Reply:
Thanks! Not sure if I understand.
How can we differentiate the response type without the conditional type inference?

Reply:
Hi @shellbj, sorry I got confused about this comment, but here we basically use the previous existing type as the default type, so for the existing/legacy code which does not provide any generic type, it would work just the same and get the same return type.

Or do you prefer to raise a breaching change PR, so don't have to consider the backward compatibility?

Reply:
Like in my above comment, please reference the Wreck `Response` type seen here instead of creating our own: https://github.com/hapijs/wreck/blob/master/lib/index.d.ts#L291-L297

Reply:
@tuckbick please notice that the Wreck `Response` is used as the return type for `get`, `post`, `patch`, `delete` (https://github.com/hapijs/wreck/blob/master/lib/index.d.ts#L83-L131), rather than `request` and `read`, which are used by service-client.

Plus, besides reusing the same `Http.IncomingMessage` as Wreck `request`, the `payload` attribute is inserted by service-client. I don't think it is reasonable to reuse types from Wreack here.